### ISSUE TITLE
carbon kept running but stopped listening on it's tcp port

### ISSUE THREAD

**Issue by Dieterbe** [graphite-project/carbon#214]
not sure how to reproduce,
but it happened to me that carbon process kept running, yet stopped listening on 2003.

i checked console.log.2014_2_7 (date it stopped) and it only contains lines like `07/02/2014 00:28:42 :: /opt/graphite/conf/storage-aggregation.conf not found, ignoring.
` (unrelated issue)

listener.log.2014_2_7 only contains lines matching `(closed cleanly|established|invalid line|Connection to the other side was lost)`

so basically nothing useful in logs. nothing useful in dmesg either.
